internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -0,0 +1,671 @@
+		if err := controllerutil.SetControllerReference(instance, secret, r.Scheme); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return secret, mutateFn, nil
+}
+
+func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(instance *adiiov1alpha1.AgentServiceConfig, host string) (*appsv1.Deployment, controllerutil.MutateFn) {
+	maxUnavailable := intstr.FromString("50%")
+	maxSurge := intstr.FromString("100%")
+
+	serviceEnv := []corev1.EnvVar{
+		{Name: "SERVICE_BASE_URL", Value: host},
+
+		// database
+		newSecretEnvVar("DB_HOST", "db.host", databaseSecretName),
+		newSecretEnvVar("DB_NAME", "db.name", databaseSecretName),
+		newSecretEnvVar("DB_PASS", "db.password", databaseSecretName),
+		newSecretEnvVar("DB_PORT", "db.port", databaseSecretName),
+		newSecretEnvVar("DB_USER", "db.user", databaseSecretName),
+
+		// image overrides
+		{Name: "AGENT_DOCKER_IMAGE", Value: AgentImage()},
+		{Name: "CONTROLLER_IMAGE", Value: ControllerImage()},
+		{Name: "INSTALLER_IMAGE", Value: InstallerImage()},
+		{Name: "SELF_VERSION", Value: ServiceImage()},
+
+		{Name: "ISO_IMAGE_TYPE", Value: "minimal-iso"},
+		{Name: "S3_USE_SSL", Value: "false"},
+		{Name: "LOG_LEVEL", Value: "info"},
+		{Name: "LOG_FORMAT", Value: "text"},
+		{Name: "INSTALL_RH_CA", Value: "false"},
+		{Name: "REGISTRY_CREDS", Value: ""},
+		{Name: "AWS_SHARED_CREDENTIALS_FILE", Value: "/etc/.aws/credentials"},
+		{Name: "DEPLOY_TARGET", Value: "k8s"},
+		{Name: "STORAGE", Value: "filesystem"},
+		{Name: "ISO_WORKSPACE_BASE_DIR", Value: "/data"},
+		{Name: "ISO_CACHE_DIR", Value: "/data/cache"},
+
+		// from configmap
+		{Name: "ADMIN_USERS", Value: "ercohen,lgamliel,mfilanov,ygoldber"},
+		{Name: "AUTH_TYPE", Value: "none"},
+		{Name: "BASE_DNS_DOMAINS", Value: ""},
+		{Name: "CHECK_CLUSTER_VERSION", Value: "False"},
+		{Name: "CREATE_S3_BUCKET", Value: "False"},
+		{Name: "ENABLE_KUBE_API", Value: "True"},
+		{Name: "ENABLE_SINGLE_NODE_DNSMASQ", Value: "True"},
+		{Name: "IPV6_SUPPORT", Value: "False"},
+		{Name: "JWKS_URL", Value: "https://api.openshift.com/.well-known/jwks.json"},
+		{Name: "OCM_BASE_URL", Value: "https://api-integration.6943.hive-integration.openshiftapps.com"},
+		{Name: "OPENSHIFT_VERSIONS", Value: `{"4.6":{"display_name":"4.6.16","release_image":"quay.io/openshift-release-dev/ocp-release:4.6.16-x86_64","rhcos_image":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.6/4.6.8/rhcos-4.6.8-x86_64-live.x86_64.iso","rhcos_version":"46.82.202012051820-0","support_level":"production"},"4.7":{"display_name":"4.7.2","release_image":"quay.io/openshift-release-dev/ocp-release:4.7.2-x86_64","rhcos_image":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.7/4.7.0/rhcos-4.7.0-x86_64-live.x86_64.iso","rhcos_version":"47.83.202102090044-0","support_level":"production"}}`},
+		{Name: "PUBLIC_CONTAINER_REGISTRIES", Value: "quay.io,registry.svc.ci.openshift.org"},
+		{Name: "SKIP_CERT_VERIFICATION", Value: "True"},
```

Comment:
:eyes:

Reply:
@carbonin had a couple of issues (https://issues.redhat.com/browse/MGMT-4740 and https://issues.redhat.com/browse/MGMT-4273) as related to the follow-up work https://issues.redhat.com/browse/MGMT-4739. I figured this would be addressed with one or both of those existing issues.

However, I think now is a great time to answer if this is addressed by an existing ticket or we need a new one.

Reply:
I think this should be handled by https://issues.redhat.com/browse/MGMT-4740

I'm worried though if we generate a self-signed cert that this will still need to be true, but we can address that in a follow-up.

Reply:
> I'm guessing the agent won't verify the REST API cert?

I think that's correct.

Reply:
added a todo and https://issues.redhat.com/browse/MGMT-4906 as a catch-all for the existing todos in the controller.